examples/templates/docker-devcontainer/main.tf

```diff
@@ -0,0 +1,372 @@
+terraform {
+  required_providers {
+    coder = {
+      source = "coder/coder"
+    }
+    docker = {
+      source = "kreuzwerker/docker"
+    }
+  }
+}
+
+locals {
+  username = data.coder_workspace_owner.me.name
+
+  # Use a workspace image that supports rootless Docker
+  # (Docker-in-Docker) and Node.js.
+  workspace_image = "codercom/enterprise-node:ubuntu"
+}
+
+variable "docker_socket" {
+  default     = ""
+  description = "(Optional) Docker socket URI"
+  type        = string
+}
+
+data "coder_parameter" "repo_url" {
+  type         = "string"
+  name         = "repo_url"
+  display_name = "Git Repository"
+  description  = "Enter the URL of the Git repository to clone into your workspace. This repository should contain a devcontainer.json file to configure your development environment."
+  default      = "https://github.com/coder/coder"
+  mutable      = true
+}
+
+provider "docker" {
+  # Defaulting to null if the variable is an empty string lets us have an optional variable without having to set our own default
+  host = var.docker_socket != "" ? var.docker_socket : null
+}
+
+data "coder_provisioner" "me" {}
+data "coder_workspace" "me" {}
+data "coder_workspace_owner" "me" {}
+
+resource "coder_agent" "main" {
+  arch            = data.coder_provisioner.me.arch
+  os              = "linux"
+  startup_script  = <<-EOT
```

Comment:
Would it make sense to extract this to a separate file? It's a pretty beefy script.